package/MDAnalysis/lib/formats/libdcd.pyx

```diff
@@ -0,0 +1,414 @@
+
+        Raises
+        ------
+        IOError
+            If you seek for more frames than are available or if the
+            seek fails (the low-level system error is reported).
+        """
+        if frame >= self.n_frames:
+            raise IOError('Trying to seek over max number of frames')
+        self.reached_eof = False
+
+        cdef fio_size_t offset
+        if frame == 0:
+            offset = self.header_size
+        else:
+            offset = self.header_size + self.firstframesize + self.framesize * (frame - 1)
+
+        ok = fio_fseek(self.fp, offset, _whence_vals["FIO_SEEK_SET"])
+        if ok != 0:
+            raise IOError("DCD seek failed with system errno={}".format(ok))
+        self.current_frame = frame
+
+    def _write_header(self, remarks, int n_atoms, int starting_step, 
+                      int ts_between_saves, double time_step):
+
+        if not self.is_open:
+            raise IOError("No file open")
+
+        if not self.mode=='w':
+            raise IOError("Incorrect file mode for writing.")
+
+        #cdef char c_remarks
+        cdef int len_remarks = 0
+        cdef int with_unitcell = 1
+
+        if isinstance(remarks, six.string_types):
+            try:
+                remarks = bytearray(remarks, 'ascii')
+            except UnicodeDecodeError:
+                remarks = bytearray(remarks)
+
+        ok = write_dcdheader(self.fp, remarks, n_atoms, starting_step, 
+                             ts_between_saves, time_step, with_unitcell, 
+                             self.charmm)
+        if ok != 0:
+            raise IOError("Writing DCD header failed: {}".format(DCD_ERRORS[ok]))
+
+    def write(self, xyz, double [:] box, int step, int natoms,
+              int ts_between_saves, int charmm, double time_step, remarks):
+        """write one frame into DCD file.
+
+        Parameters
+        ----------
+        xyz : ndarray, shape=(n_atoms, 3)
+            cartesion coordinates
+        box : ndarray, shape=(3, 3)
+            Box vectors for this frame
+        step : int
+            current step number, 1 indexed
+        time : float
+            current time
+        natoms : int
+            number of atoms in frame
+
+        Raises
+        ------
+        IOError
+        """
+        if not self.is_open:
+            raise IOError("No file open")
+        if self.mode != 'w':
+            raise IOError('File opened in mode: {}. Writing only allowed '
+                               'in mode "w"'.format('self.mode'))
+
+        #cdef double [:,:] unitcell = box
+        cdef FLOAT_T[::1] x = xyz[:, 0]
+        cdef FLOAT_T[::1] y = xyz[:, 1]
+        cdef FLOAT_T[::1] z = xyz[:, 2]
+
+	# prerequisite is a file struct for which the dcd header data
+	# has already been written
+        self._write_header(remarks=remarks, n_atoms=xyz.shape[0], starting_step=step,
+                           ts_between_saves=ts_between_saves,
+                           time_step=time_step)
```

Comment:
I don't understand this. Do we write the header for every frame now? I understood the old dcd code that this should only be written once.